.cirrus.yml

```diff
@@ -0,0 +1,78 @@
+_pipeline: &pipeline
+  info_script: |
+    meson --version
+    ninja --version
+  submodule-update_script: git submodule update --init --recursive
+  configure_script: meson build
+  compile_script: ninja -C build
+  test_script: ninja -C build test
+
+Debian (gcc)_task:
+  container:
+    image: snaipe/ci-meson:debian-10
+  setup_script: |-
+    apt-get update
+    apt-get install -y pkg-config cmake libgit2-dev libffi-dev libnanomsg-dev
+    pip3 install cram==0.7
+  <<: *pipeline
+
+Alpine (gcc,x86_64)_task:
+  container:
+    image: snaipe/ci-meson:alpine
+  setup_script: &alpine-deps |-
+    apk add --no-cache cmake libgit2-dev libffi-dev
+    pip3 install cram==0.7
+  <<: *pipeline
+
+Alpine (gcc,i386)_task:
+  container:
+    image: snaipe/ci-meson:alpine-x86
+  setup_script: *alpine-deps
+  <<: *pipeline
+
+Alpine (gcc,arm)_task:
+  container:
+    image: snaipe/ci-meson:alpine-armhf
+  env:
+    CIRRUS_SHELL: /sbin/qemu-sh
+  setup_script: *alpine-deps
+  <<: *pipeline
+
+Alpine (gcc,aarch64)_task:
+  container:
+    image: snaipe/ci-meson:alpine-aarch64
+  env:
+    CIRRUS_SHELL: /sbin/qemu-sh
+  setup_script: *alpine-deps
+  <<: *pipeline
+
+MacOS_task:
+  osx_instance:
+    image: mojave-xcode-10.2
+  env:
+    PKG_CONFIG_PATH: ${PKG_CONFIG_PATH}:/usr/local/opt/libffi/lib/pkgconfig
+  setup_script: |-
+    brew install meson cmake pkg-config libffi libgit2
+    pip3 install cram==0.7
+  <<: *pipeline
+
+FreeBSD_task:
+  freebsd_instance:
+    image_family: freebsd-12-0
+  setup_script: |-
+    pkg install -y py37-pip meson git libffi libgit2
+    pip3 install cram==0.7
+  <<: *pipeline
+
+Windows_task:
+  windows_container:
+    image: cirrusci/windowsservercore:cmake
+    os_version: 2019
+  env:
+    PATH: C:\\Python;C:\\Python\\Scripts;C:\\ProgramData\\chocolatey\\lib\\ninja\\tools;%PATH%
+  setup_script: |-
+    choco install -y --no-progress python3 --params "/InstallDir:C:\Python"
+    choco install -y --no-progress ninja cmake
+    pip install meson cram==0.7
```

Comment:
meson setup seems to be consume some minutes in BoxFort: https://cirrus-ci.com/task/5856113753063424,
maybe it could be added to the [cache](https://cirrus-ci.org/guide/writing-tasks/#cache-instruction) (with sane fingerprint).

Reply:
Yeah, I hear you. I've had mixed results with getting the cache to somewhere that didn't break previous builds... one of the most problematic aspects was when I left my cirrus-ci WIP sit for a few weeks alone, and then re-triggering a build, only to find out that I got an older dependency cached while the underlying image moved forward.

For the scope of this PR, I think I'll leave the cache alone, and just make this "slow" but correct. We can optimize build times further when we get back to a stable build and CI environment.